java-util/src/main/java/io/druid/java/util/common/parsers/Parser.java

```diff
@@ -35,6 +35,11 @@
   public Map<K, V> parse(String input);
 
   /**
+   * Resets state within a parser.
+   */
+  public void reset();
```

Comment:
The javadoc for this method should say when it's going to be called: something like "at the start of each new file after the first one". It should also specify if it's going to be called before the very first call to `parse` or not. IMO, we shouldn't promise anything in particular, and say something like: "may or may not be called before the very first file"